mlir/lib/Quantum/Utils/QuantumSplitting.cpp

```diff
@@ -89,17 +91,73 @@ void AugmentedCircuitGenerator::generate(Region &region, OpBuilder &builder)
         else if (auto extractOp = dyn_cast<quantum::ExtractOp>(op)) {
             cacheDynamicWire(extractOp, builder);
         }
-        else if (auto gate = dyn_cast<quantum::DifferentiableGate>(op)) {
-            ValueRange diffParams = gate.getDiffParams();
-            if (!diffParams.empty()) {
-                for (Value param : diffParams) {
-                    builder.create<ListPushOp>(gate.getLoc(), oldToCloned.lookupOrDefault(param),
-                                               cache.paramVector);
+        else if (auto gate = dyn_cast<quantum::ParametrizedGate>(op)) {
+            ValueRange params = gate.getAllParams();
+
+            for (Value param : params) {
+                Location loc = gate.getLoc();
+                Value clonedParam = oldToCloned.lookupOrDefault(param);
+                Type paramType = clonedParam.getType();
+
+                if (!paramType.dyn_cast<RankedTensorType>()) {
```

Comment:
I would use `isa` if the casted value is not needed. Also I think this form has been deprecated and the global functions should be used:
```suggestion
                if (!isa<RankedTensorType>(paramType)) {
```